api/types/types.go

```diff
@@ -470,6 +470,7 @@ type ServiceInterface struct {
 	EnableTls                bool                     `json:"enableTls,omitempty"`
 	TlsCredentials           string                   `json:"tlsCredentials,omitempty"`
 	PublishNotReadyAddresses bool                     `json:"publishNotReadyAddresses,omitempty"`
+	Namespace                string                   `json:"namespace,omitempty" yaml:"namespace,omitempty"`
```

Comment:
My line thought was that by specifying a determined namespace from where you want to expose a deployment/statefulset, all targets would be within this same namespace. Hence, since the namespace would be the same for all targets I left the new attribute in the upper-level struct (ServiceInterface).

Reply:
Is there a scenario where you could spot the namespace being different for a specific target?

Reply:
If I create a service, using `skupper service create` and pass a target namespace as an argument,
my perception (as a user) is that the service itself will be created in that target namespace.

But if I am specifying the target namespace at the time I am binding a target (or while exposing one),
then it feels like the target is located in that namespace.

So it is an usability issue to me, as it makes more sense to associate the namespace with the target being bound.

Reply:
Moved the attribute to ServiceInterfaceTarget and also removed `--target-namespace` flag from `ServiceCreate` command, since the idea of the PR is to expose a target located in a different namespace rather than create the service in another namespace. Besides that, I've added more tests to the overall changes introduced by this PR